test/jdk/jdk/internal/util/ArchTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import jdk.internal.util.Architecture;
+
+import static jdk.internal.util.Architecture.AARCH64;
+import static jdk.internal.util.Architecture.ARM;
+import static jdk.internal.util.Architecture.IA64;
+import static jdk.internal.util.Architecture.PPC64LE;
+import static jdk.internal.util.Architecture.RISCV64;
+import static jdk.internal.util.Architecture.S390X;
+import static jdk.internal.util.Architecture.X64;
+import static jdk.internal.util.Architecture.X86;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * @test
+ * @bug 8304915
+ * @summary Verify Architecture enum maps to system property os.arch
+ * @modules java.base/jdk.internal.util
+ * @run junit ArchTest
+ */
+public class ArchTest {
+    /**
+     * Test consistency of System property "os.arch" with Architecture.current().
+     */
+    @Test
+    public void nameVsCurrent() {
+        String osArch = System.getProperty("os.arch").toLowerCase(Locale.ROOT);
```

Comment:
Is it planned to determine os.arch with the same enum in the future (keeping the legacy names)?

Reply:
That's a different set of twisty build and native source files, I'll create an issue to re-visit that.
It may not change too much, the native code needs to initialize `os.arch` very early.
[JDK-8305676](https://bugs.openjdk.org/browse/JDK-8305676)